### DISCUSSION TITLE
Many to One return value

### DISCUSSION THREAD

**Discussion by oleva** [directus/directus#16480] [Archive]
Hi,

Trying to learn Directus and really digging it so far!
This might be a silly question but tried to search for an answer and couldn't find one.

I'm using SDK to get items. I have created Many to One field for directus_users. It's related to "company" collection that has UUID.

Can I determine what field Directus returns? Now when reading directus_users, it gives the companys UUID, I'd like it to return the companys name. I only need the one field from the company collection, so a new query just for the name is kinda of an overkill. I thought that I could use the display template for that, but apparently it doesn't work that way.

Thanks guys!

**Comment by domdus** [comment]
Check out the `fields` param/property also available in the SDK.

Example:
`fields: ["id,company.name"]`

Retrieve all fields from certain levels deep
`fields: ["*"]`
`fields: ["*.*.*"]`
or specific fields
`fields: ["id,status,title,related_item.*"]`

In SDK:
const result = await directus.items("directus_users").readByQuery({
    fields: ["id,status,company.name"],
   ....
});

https://docs.directus.io/reference/sdk.html

**Reply by oleva** [reply]
Works like a charm. Thanks!

**Reply by danilowm** [reply]
thanks, works in new sdk too


```
const result = await client.request(
   readItems("Voucher", {
      filter: {
         status: { _eq: "published" },
      },
      fields: ["*", "Loja.Nome"],
   })
);
```

**Comment by rijkvanzanten** [comment]
> This might be a silly question but tried to search for an answer and couldn't find one.

I'm curious! What keywords did you search for? We get this question a lot, so seemingly people aren't finding their way to https://docs.directus.io/reference/introduction.html#relational-data organically 🤔

**Reply by oleva** [reply]
Well, IMO some more examples would cut it. Mainly this is pretty short: https://docs.directus.io/reference/sdk.html#fields
Didn't find much either on Stack overflow. In general, I think there could just be better examples for a bunch of things. Also, the use of wildcards isn't documented (or couldn't find it again). 

For me one of the biggest things and reasons to use Directus is the ability to give custom permissions to roles. For example, the field validation rule can be set to $CURRENT_USER.user_company is a huge thing for data security and scalability. This kind of a thing is also hard to find in the docs.

Maybe some kind of tips & tricks section?

Lots of love for the platform though, been at it for a few days now.

**Reply by rijkvanzanten** [reply]
Cool cool! Glad to hear that our current focus on adding a "cookbook" to the docs with common "recipes" for this type of thing is the right direction 🙂 https://github.com/directus/docs/pull/197 (cc @erondpowell)

**Reply by erondpowell** [reply]
> ... so seemingly people aren't finding their way

> ...I think there could just be better examples for a bunch of things.

I have some ideas to improve all of this, stay tuned 🎵 

@oleva Can you tell us more about your use case? 
Perhaps you have a good recipe here for the cookbook.

**Comment by raselmiazi** [comment]
```json
{
        "type": "uuid",
        "meta": {
          "interface": "file-image",
          "special": [
            "file"
        ]
        },
        "field": "Image_preview"
      },
      {
        "collection": "Image_themes",
        "field": "Image_preview",
        "related_collection": "directus_files",
        "meta": {
          "sort_field": null
        },
        "relationship": {
          "type": "many-to-one",
          "on_delete": "SET NULL"
        }
      }
```

i am writing sdk import script but relationship properly not work  please any help me

**Reply by raselmiazi** [reply]
![image](https://github.com/directus/directus/assets/163228495/67975148-13c4-4132-aee9-8e790c3228b6)
![image](https://github.com/directus/directus/assets/163228495/eab1193e-52fa-4c67-a642-de373bb53ab7)

**Comment by raselmiazi** [comment]
![Screenshot 2024-06-20 155206](https://github.com/directus/directus/assets/163228495/b26a4f1a-55fe-4ddd-8859-db3d31adb0b6)
